### PR TITLE
[Proxmox] Add ability to select CPU type

### PR DESCRIPTION
**Pull request by brickstool**
The only CPU type used in Proxmox VMs generated by Packer is `host`. While this improves performance, it can create issues when live migrating VMs between Proxmox hosts with different CPU types (e.g. Intel to AMD or vice-versa) by causing the qemu process to stop (see the second paragraph of 'CPU Type' from the [Proxmox Wiki](https://pve.proxmox.com/wiki/Qemu/KVM_Virtual_Machines#qm_virtual_machines_settings)).

This PR adds the ability to set the CPU type through the `cpu_type` option for the Proxmox builder. Additionally, it changes the default CPU type from `host` to `kvm64` as this is the default used by Proxmox.

### CODE REVIEW

**Comment by SwampDragons** [comment]
@carlpett can you review?  :D

**Comment by carlpett** [comment]
@SwampDragons FYI, according to the CODEOWNERS docs, if you grant write access I should be automatically tagged on these

**Comment by SwampDragons** [comment]
Done :)

**Comment by carlpett** [comment]
@brickstool Looks like you got a whitespacing conflict with your other PR, apart from that, good to go!

File: builder/proxmox/config.go [hunk 1]
```diff
@@ -39,6 +39,7 @@ type Config struct {
 
 	Memory         int          `mapstructure:"memory"`
 	Cores          int          `mapstructure:"cores"`
+  CPUType        string       `mapstructure:"cpu_type"`
```
Comment (azr) [inline builder/proxmox/config.go]: Woops; I failed fixing this 🙂 
```suggestion
	CPUType        string       `mapstructure:"cpu_type"`
```

Comment (azr) [reply builder/proxmox/config.go]: Weird; I was allowed to fix the conflict; but not to commit this.

**Comment by brickstool** [comment]
Thanks @SwampDragons, @carlpett and @azr! This PR and the other Proxmox PR are my first two PR's ever, so thanks for accepting and helping out!

**Comment by SwampDragons** [comment]
@brickstool thanks for contributing!

**Pull request merged**